toucan_connectors/oauth2_connector/oauth2connector.py

```diff
@@ -66,13 +70,15 @@ def __init__(
         self.token_url = token_url
         self.redirect_uri = redirect_uri
 
+    def _check_secret_keeper_exists(self) -> None:
+        if self.secrets_keeper is None:
+            raise SecretKeeperMissingError("Secret keeper is not set on oauth2 connector.")
```

Comment:
```suggestion
    def _secrets_keeper(self) -> SecretsKeeper:
        if self.secrets_keeper is None:
            raise SecretKeeperMissingError("Secret keeper is not set on oauth2 connector.")
        return self.secrets_keeper
```

It'd you to do `self._secrets_keeper().something` everywhere in the code